File: 24swapNodesInPairs/swapNodesInPairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    const result = new ListNode(null)
    let current = result
    current.next = head
    let a = null
    while ((a = current.next) && current.next.next)
    {
        current.next = curren.next.next
        a.next = a.next.next
        current.next.next = a
        current = a
    }
    return result.next
};

File: 24swapNodesInPairs/swapNodesInPairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        result = current = ListNode(None)
        result.next = head
        while (a := current.next) and current.next.next:
            current.next = current.next.next
            a.next = a.next.next
            current.next.next = a
            current = a
        return result.next
            
